internal/gatewayapi/translator.go

```diff
@@ -210,7 +223,9 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
 			default:
 				protocol = string(listener.Protocol)
 			}
-			portListenerInfo[listener.Port].protocols.Insert(protocol)
+			if protocol != string(v1beta1.UDPProtocolType) {
```

Comment:
why is this needed ?

Reply:
Because this part  of code(compatible protocol, hostname verification) is only for TCP.

Reply:
BTW, ProcessListeners method is growing too large. Should we refacory it to make it easier to read?

Reply:
sure, can you raise a GH issue so we can track and tackle in the near future ?

Reply:
The original check if only for layer-7(HTTP, HTPS, TLS), I refactory this method and add layer-4(TCP UDP) port conflicts check.

Reply:
> sure, can you raise a GH issue so we can track and tackle in the near future ?

I alreay splited [ProcessListeners](https://github.com/envoyproxy/gateway/pull/728/commits/6b87db0239b06c030e57be6991eddd49e47c2999) into serveral smaller methods since it has grown too large and make it hard to add UDP related code.

Reply:
> sure, can you raise a GH issue so we can track and tackle in the near future ?

There are still some duplications in the gateway API translator, but it's better to do the refractory in another PR since this one is already quite complex. I created an issue to track it. https://github.com/envoyproxy/gateway/issues/736